src/parallel/ParticleDataFull.cpp

```diff
@@ -60,9 +60,9 @@ void ParticleDataFull::MoleculeToParticleData(ParticleDataFull &particleStruct,
 	particleStruct.D[2] = molecule.D(2);
 }
 
-void ParticleDataFull::ParticleDataToMolecule(const ParticleDataFull &particleStruct, Molecule &molecule) {
+Molecule ParticleDataFull::ParticleDataToMolecule(const ParticleDataFull &particleStruct) {
```

Comment:
Out of interest: Is the whole object returned or just a pointer?
If former, isn't this quite bad since a Molecule object is rather big?